frontend/src/components/Utils/filter-input.test.js

```diff
@@ -0,0 +1,103 @@
+import * as firebase from 'firebase/app';
+import 'firebase/firebase-firestore';
+
+import { getUidFromUserEmail } from './temp-auth-utils';
+import { getCleanedTextInput, getTimestampFromDateString,
+         getCollaboratorUidArray }  from './filter-input.js';
+
+describe('getCleanedTextInput tests', () => {
+  test('No input entered in form (empty string)', () => {
+    const testDefaultValue = 'Untitled Trip';
+    const testRawName = '';
+    const expectedTripName = testDefaultValue;
+
+    const testTripName = getCleanedTextInput(testRawName, testDefaultValue);
+
+    expect(testTripName).toEqual(expectedTripName);
+  });
+
+  test('Input entered into form', () => {
+    const testDefaultValue = 'Untitled Trip';
+    const testRawName = 'Trip to No Man\'s Land';
+    const expectedTripName = testRawName;
+
+    const testTripName = getCleanedTextInput(testRawName, testDefaultValue);
+
+    expect(testTripName).toEqual(expectedTripName);
+  });
+});
+
+const mockTimeNow = 0;
+jest.mock('firebase/app', () => ({
+    firestore: {
+      Timestamp: {
+          now: () => mockTimeNow,
+          fromDate: (date) => date
+      }
+    }
+}));
+describe('getTimeStampFromDateString tests', () => {
+  test('No date entered in form', () => {
+    const expectedTimestamp = mockTimeNow;
+    const testRawDate = '';
+
+    const testTimestamp = getTimestampFromDateString(testRawDate);
+
+    expect(testTimestamp).toEqual(expectedTimestamp);
+  });
+
+  test('Date entered in form', () => {
+    const testDate = new Date(2020, 5, 4); // July 4, 2020
```

Comment:
It is a [month index](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/Date) (0 indexed). That is why I left the comment because I thought it was 1 index in the constructor.